app/src/main/java/org/kiwix/kiwixmobile/data/local/KiwixDatabase.java

```diff
@@ -124,21 +127,44 @@ protected boolean onUpgrade(SQLiteDatabaseWrapper db, int oldVersion, int newVer
     }
     if (newVersion >= 13) {
       tryAddColumn(BookDatabaseEntity.NAME);
-      tryAddColumn(Bookmarks.ZIM_NAME);
+      tryAddColumn(Bookmark.ZIM_NAME);
     }
     if (newVersion >= 14 && oldVersion < 14) {
       tryDropTable(BookDatabaseEntity.TABLE);
       tryCreateTable(BookDatabaseEntity.TABLE);
     }
+    if (newVersion >= 15 && oldVersion < 15) {
+      tryAddColumn(Bookmark.ZIM_FILE_PATH);
+      tryAddColumn(Bookmark.FAVICON);
+      migrateBookmarksVersion15();
+    }
     return true;
   }
 
+  private void migrateBookmarksVersion15() {
```

Comment:
@mhutti1 This is how I am migrating the DB. But the length of the list of books I'm getting from the DB is zero. :/
I'm not able to figure out why. Can you please have a look? :)

Reply:
Possibly the change in Table name?

Reply:
I doubt it though.

Reply:
Ok. I think its because: https://github.com/kiwix/kiwix-android/pull/791/files/44f4371f0568bfd3946b6de1a686a846f23e5927#diff-169d649a310e70bf3e0464a9794f02bbR133